apps/kyb-app/src/components/layouts/AppShell/Logo.tsx

```diff
@@ -16,14 +17,16 @@ const prefetchImage = (url: string) =>
 const fallback = (timeout: number) => new Promise(resolve => setTimeout(resolve, timeout));
 
 export const Logo = ({ logoSrc, appName, onLoad }: Props) => {
+  const onLoadRef = useRefValue(onLoad);
+
   useEffect(() => {
     if (!onLoad) {
       return;
     }
 
     // Using race here in case if image is corrupted or load takes to long we don't want to lock stepper breadcrumbs forever.
-    Promise.race([prefetchImage(logoSrc), fallback(3000)]).then(onLoad);
-  }, [logoSrc, onLoad]);
+    Promise.race([prefetchImage(logoSrc), fallback(3000)]).then(onLoadRef.current);
+  }, [logoSrc, onLoadRef]);
```

Comment:
Should this line be:
```suggestion
  }, [logoSrc, onLoadRef.current]);
```

Reply:
Nope, it will trigger useEffect each time prop updates, that was the case of bug. So only using link to callback and not callback itself as dependency.